app/pages/lab-pages-editor/components/TasksPage/components/NewTaskButtonAndDialog.jsx

```diff
@@ -0,0 +1,71 @@
+/* eslint-disable no-console */
+/* eslint-disable react/react-in-jsx-scope */
+/* eslint-disable react/require-default-props */
+/* eslint-disable radix */
+/* eslint-disable react/jsx-boolean-value */
+/* eslint-disable react/forbid-prop-types */
+/* eslint-disable react/no-unknown-property */
+
+import { useRef } from 'react';
+
+import CloseIcon from '../../../icons/CloseIcon.jsx';
+// import strings from '../../../strings.json'; // TODO: move all text into strings
+
+function NewTaskButtonAndDialog() {
+  const newTaskDialog = useRef(null);
+
+  function openNewTaskDialog() {
+    newTaskDialog.current?.showModal();
+  }
+
+  function closeNewTaskDialog() {
+    newTaskDialog.current?.close();
+  }
+
+  /*
+  WARNING: the <dialog> element should automatically close with the Esc key,
+  but this seems disabled. This doesn't seem to be a problem with React, but
+  with PFE.
+   */
+  // function dialogOnKeyUp(e) {
+  //  if (e.key === 'Escape') newTaskDialog.current?.close();
+  // }
+
+  return (
+    <>
+      <button
+        className="flex-item big primary"
+        onClick={openNewTaskDialog}
+        type="button"
+      >
+        Add a new Task +
+      </button>
+      <dialog
+        autofocus="true"
```

Comment:
```suggestion
        autoFocus
```
This is why you're getting ESLint warnings.

Reply:
Don't focus the dialog itself by the way. 

Focus the first focusable element or the first heading. 

The [WAI-ARIA dialog example](https://www.w3.org/WAI/ARIA/apg/patterns/dialog-modal/examples/dialog/) has some discussion of this.

Reply:
Is this dialog supposed to be modal, by the way?

Reply:
Ignore me, I just saw that you're using `showModal()`. 👍

Reply:
The autofocus thing here is an odd one - I think I'll just delete it altogether since adding/removing `autofocus="true"` **has no effect** on the UI/UX (on macOS + Chrome 119/FF 119/Safari 15). i.e. the first interactable element (i.e. the close button) _always_ gets focus on showModal() anyway. 🤨 

Other misc dev notes:
- `autofocus="true"` in React renders as `<dialog autofocus="true">...</dialog>` in the HTML
- `autofocus`, `autofocus={true}`, `autoFocus`, `autoFocus={true}`, and `autoFocus="true"` will all just render as `<dialog>...</dialog>`
- I wondered if this is just a quirk with me adding autofocus to `<dialog>` instead of a proper interactable element, so I tried...
    ```
    <dialog>
      <input type="text" defaultValue="this is just a distraction" />
      <input type="text" defaultValue="this should get focus when modal opens" autofocus="true" />
    </dialog>
    ```
- ...and the ONLY time that the second text input gets focus on .showModal() is with `autofocus="true"`. autoFocus doesn't work, despite [React saying it should.](https://react.dev/reference/react-dom/components/input#props)
- Maybe this is a quirk with React 17? I don't know, but I'm making a note of it just in case.